apps/www/src/components/root/SearchBar.tsx

```diff
@@ -2,32 +2,34 @@
 
 import { Button, Input } from "@nextui-org/react"
 import { useSearchParams } from "next/navigation"
-import { useState } from "react"
+import { FormEvent, useState } from "react"
 import { FaSearch } from "react-icons/fa"
 import { useRouter } from "next-nprogress-bar"
 
 export default function SearchBar() {
     const router = useRouter()
     const [input, setInput] = useState(useSearchParams().get("q") ?? "")
-    const submit = () => router.push(input.length ? `/?q=${input}` : "/")
+    const submit = (e: FormEvent) => {
+        e.preventDefault()
+        router.push(input.length ? `/?q=${input}` : "/")
+    }
 
     return (
-        <div className="flex w-full items-center gap-2">
+        <form className="flex w-full items-center gap-2" onSubmit={submit}>
             <Input
                 className="ml-auto sm:w-fit"
                 classNames={{ inputWrapper: "border rounded-full" }}
                 value={input}
                 onChange={(e) => setInput(e.target.value)}
-                onKeyDown={(e) => e.key === "Enter" && submit()}
                 type="search"
                 placeholder="Search..."
                 labelPlacement="outside"
                 size="sm"
                 startContent={<FaSearch className="mx-1" />}
             />
-            <Button onClick={submit} className="h-8 rounded-full" color="primary">
+            <Button type="submit" onClick={submit} className="h-8 rounded-full" color="primary">
```

Comment:
`onClick` event handler isn't needed when button type is `submit`.

Reply:
yes correct, let me remove it.